src/main/java/buildcraftAdditions/client/gui/gui/GuiFluidicCompressor.java

```diff
@@ -85,8 +67,45 @@ public String getInventoryName() {
 	}
 
 	@Override
+	public void widgetActionPerformed(WidgetBase widget) {
+		if (widget instanceof WidgetButtonFluidicCompressor) {
+			fluidicCompressor.fill = ((WidgetButtonFluidicCompressor) widget).fill;
+			PacketHandler.instance.sendToServer(new MessageFluidicCompressorA(fluidicCompressor.fill, fluidicCompressor));
+		}
+	}
+
+	@Override
 	public void initialize() {
-		addWidget(new WidgetFluidTank(0, (width - xSize) / 2 + 53, (height - ySize) / 2 + 16, 16, 52, this, fluidicCompressor.tank));
+		addWidget(new WidgetFluidTank(0, guiLeft + 53, guiTop + 16, 16, 52, this, fluidicCompressor.tank));
+		addWidget(new WidgetButtonFluidicCompressor(1, guiLeft + 20, guiTop + 34, 19, 16, this, fluidicCompressor.fill));
+	}
+
+	public static class WidgetButtonFluidicCompressor extends WidgetBase {
```

Comment:
There is a WidgetButtonUpdate that would perfectly fit the needs for this? why re-invent the wheel